src/main/java/org/openrewrite/java/spring/ReplaceRequiredAnnotationOnSetterWithAutowired.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.*;
+import org.openrewrite.java.search.FindAnnotations;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+
+import java.util.Comparator;
+
+public class ReplaceRequiredAnnotationOnSetterWithAutowired extends Recipe {
```

Comment:
Instead of a recipe would this also be possible with a declarative recipe `ChangeType`? Or are there edge cases that won't cover?

Whenever possible try to do declaratively what we can; that helps reduce what we need to maintain going forward.

Reply:
Good suggestion, Thanks!